components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -136,6 +143,14 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
         return true;
     }
 
+    @Override
+    public OAuth2AccessTokenRespDTO issue(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        tokReqMsgCtx.setTokenBinding(tokenDO.getTokenBinding());
+        revokeExistingToken(tokenDO.getConsumerKey(), tokenDO.getAccessToken());
```

Comment:
Revoking the existing token might slow down the login process. 
(I am assuming this call happens right after login or during login most of the times)

Reply:
Can't we use the token state id approach?
(Generating a random uuid for token state id)

Reply:
Then we have to update the entire token caching logic because we don't use the token id state when caching the token[1] and due to this reason we cannot differentiate the auth code token and org switch token from token

[1] - https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java#L152

Reply:
Shall we go ahead with this approach since we are unable to use the same binding ref with two tokens that have the same scope, consumer key, authorized user, and authenticatedIDP. If we add the same binding ref to the two tokens (Authz_code and ors_switch tokens), it will be unable to differentiate for which grant type the relevant token was issued for. So eventually the org_swich grant type will use the token issued for the authz_code grant type.

Or we can use the same token for both of these grant types, it will also solve this problem.

Reply:
Or we can use the same token for both of these grant types, it will also solve this problem.
--> can you check whether the details persisted currently for the two tokens are the same?